CRM/Upgrade/Incremental/sql/4.7.23.mysql.tpl

```diff
@@ -41,3 +41,10 @@ ON price_field.price_field_id = cpf.id
 LEFT JOIN civicrm_price_set ps ON ps.id = cpf.price_set_id
 SET cpf.is_active = 1
 WHERE ps.is_quick_config = 1 AND cpf.is_active = 0;
+
+-- CRM-20935 Clean up orphaned profile links for deleted events
+DELETE civicrm_uf_join
+FROM civicrm_uf_join
+LEFT JOIN civicrm_event e on civicrm_uf_join.entity_id = e.id
+WHERE (civicrm_uf_join.module = 'CiviEvent' OR civicrm_uf_join.module = 'CiviEvent_Additional')
+AND e.id IS NULL;
```

Comment:
I tested this manually, and also thought through this.  This SQL is exactly right, though of course it will need to be moved to 4.7.25.mysql.tpl.